nrf-hal-common/src/twis.rs

```diff
@@ -309,14 +486,80 @@ pub enum Error {
 
 #[derive(Debug, Eq, PartialEq, Clone, Copy)]
 pub enum TwiEvent {
-    Read,
+    Stopped,
+    Error,
+    RxStarted,
+    TxStarted,
     Write,
+    Read,
+}
+
+/// A DMA transfer
+pub struct Transfer<T: Instance, B> {
+    inner: Option<Inner<T, B>>,
```

Comment:
Why is this in an `Option` if it's only `None` when the `Transfer` is being destroyed? Because the `Drop` impl prevents moving out of fields? If you move the `Drop` impl to `Inner` you can move out of `Transfer`'s fields again.

Reply:
The reason was to be able to move out fields in `wait()` and i can't figure out a better way to do that... Or did i misunderstand your suggestion? :)

Reply:
`wait` takes `self` by value, so it can move out of its fields already, without the `Option` (if you apply the `Drop` suggestion)

Reply:
But it's the fields from `inner` i want to move out, which would be prevented by impl `Drop` on `Inner` no? Sorry if i'm thinking slow here, fever brain...

Reply:
Oh, right, sorry. Then maybe put a FIXME comment there that explains this?

Reply:
Well, you could have `inner` as a struct with no fields, and conjure the registers on `Drop` to be able to stop the DMA. Or do something like I did here: 
https://github.com/stm32-rs/stm32f4xx-hal/blob/36ad03a633145592c8f6bb8c651e25acfb323b08/src/dma/mod.rs#L1095-L1108

I don't have strong feelings about this though.

Reply:
Good ideas! Although to me the `Option` alternative feels like less mental overhead and it's still not too semantically off no? No strong feelings here either...